core/coreunix/add.go

```diff
@@ -470,8 +471,13 @@ func (adder *Adder) addFile(file files.File) error {
 		return err
 	}
 
+	addFileName := file.FileName()
+	if addFileName == "" && adder.Name != "" {
```

Comment:
@keks WDYT about checking for an empty filename? I think this covers the `stdin` case in CLI, I'm not sure what effect this may have in the HTTP case, that's why we're resetting the `adder.Name` option to rename only one file (in the case HTTP may send many unnamed files).

Reply:
It looks like there are several occasions where we use an empty string as file name. I just looked into this and it looks like when using stdin, te `FullName()` should be `os.Stdin.Name()`, i.e. `/dev/stdin`. IMHO that would be a better check, and should also possible on the http server side.

Reply:
hello @keks ,i want to fix this bug.And i do not understand the fullname.can u explain more for me?

Reply:
@kjzz Sure! The `file` variable is a `"files".File`, an interface described here: https://godoc.org/github.com/ipfs/go-ipfs-cmdkit/files#File. It has the method `FileName()`, which is currently used in the check. But it also has the method `FullName()`, which should return `/dev/stdin` if the file is stdin.  If you check `file.FullName() == "/dev/stdin"` or `file.FullName() == os.Stdin.Name()`, that should be more reliable, because there are other reasons why `file.FileName()` may be the empty string. I only grepped through the code in go-ipfs-cmds and saw a lot of files with empty string as name. I'm not sure this will be a problem in practice, but I can't rule out that it won't. I think checking `file.FullName()` should be much more reliable.

Reply:
Thanks @keks! That sounds much more robust, @kjzz, so the check would be
```
if file.FullName() == os.Stdin.Name() && adder.Name != "" {
```